### PR TITLE
Adds scavengers for ash walkers.

### PR DESCRIPTION
**Pull request by Cheridan**
Ash walkers have a couple problems:
Their dens get super messy
They tend to die excessively, many people choose to just suicide rather than try for heals.

So here are some friendly aliens to eat up all the mess and turn it into healing milk.

gutlunchpng

Inspired by the "tribal cleanbot".

Other stuff: Renames var/species to animal_species to differentiate it from the carbon/human species var.
Sprite reorganization: Moves old asteroid mobs to lavaland_monsters.dmi just so they're all grouped. Removes a sprite from mob.dmi.

### CODE REVIEW

File: code/modules/mob/living/simple_animal/friendly/farm_animals.dm [hunk 1]
```diff
@@ -272,28 +272,44 @@ var/global/chicken_count = 0
 	else
 		SSobj.processing.Remove(src)
 
-/obj/udder
 
-/obj/udder/New()
+/obj/item/udder
```
Comment (Cheridan) [inline code/modules/mob/living/simple_animal/friendly/farm_animals.dm]: Would this be better as a /obj/item/organ/internal or do you guys think that'd be overkill?

File: code/modules/mob/living/simple_animal/hostile/mining_mobs.dm [hunk 2]
```diff
@@ -733,6 +733,76 @@
 	icon = 'icons/obj/surgery.dmi'
 	icon_state = "demon_heart"
 
+
+//Gutlunches
+
+/mob/living/simple_animal/hostile/asteroid/scavenger/
```
Comment (KazeEspada) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs.dm]: Slash at the end of the path

File: code/modules/mob/living/simple_animal/hostile/mining_mobs.dm [hunk 3]
```diff
@@ -733,6 +733,76 @@
...
+		HandleProcreation()
+	LoseTarget()
+	..()
+
+/mob/living/simple_animal/hostile/asteroid/scavenger/proc/HandleProcreation()
+	if(udder.reagents.total_volume == udder.reagents.maximum_volume)
+		new /mob/living/simple_animal/hostile/asteroid/scavenger/(get_turf(src))
+
+/mob/living/simple_animal/hostile/asteroid/scavenger/female
+	gender = FEMALE
```
Comment (KazeEspada) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs.dm]: Why is there a female subtype?

Comment (Cheridan) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs.dm]: sexual dimorphism

Comment (KazeEspada) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs.dm]: But it has the same sprites as the males?

Comment (Cheridan) [inline code/modules/mob/living/simple_animal/hostile/mining_mobs.dm]: read the fucking tags holy shit it's a Work In Progress 
read the fucking OP: "Still need to [...] implement breeding"

**Comment by KorPhaeron** [comment]
Going to need to set their faction to include neutral so they dont freak out around the lizards

**Comment by coiax** [comment]
10/10 for tribal cleanbot memes

**Comment by Cheridan** [comment]
@KorPhaeron Their vars are set so that they don't bother people regardless of faction, unless the target is in crit. Which I thought would be kinda funny, like "geez die already."

File: code/modules/mob/living/simple_animal/friendly/farm_animals.dm (hunk 1 above)
Comment (Incoming5643) [inline code/modules/mob/living/simple_animal/friendly/farm_animals.dm]: The odd standalone nature of udders has bugged me for a long time.

**Comment by Zilenan91** [comment]
Ash Walkers = Orks
Scavengers = Gretchin

The coder conspiracy to eventually turn /tg/ into 40k is now confirmed

**Comment by Cheridan** [comment]
DELETE THIS

**Comment by Incoming5643** [comment]
Hello?

File: code/modules/mob/living/simple_animal/friendly/farm_animals.dm (hunk 1 above)
Comment (Core0verload) [inline code/modules/mob/living/simple_animal/friendly/farm_animals.dm]: Making it an internal organ that, when implanted, constantly adds it's reagent to your blood, would be nice.

**Comment by optimumtact** [comment]
Hello o7

File: code/modules/mob/living/simple_animal/friendly/farm_animals.dm (hunk 1 above)
Comment (Cheridan) [inline code/modules/mob/living/simple_animal/friendly/farm_animals.dm]: @Incoming5643 yeah it was literally the only /obj in the code.

@Core0verload That's not really how udders work...
I mean it wouldn't be hard at all to move the udder code up in order to allow udder transplants so you can milk your crewm-- nope nope nope no forget i said anything

Comment (optimumtact) [inline code/modules/mob/living/simple_animal/friendly/farm_animals.dm]: No please, do go on, you had my attention and now you have my curiousity

Comment (Core0verload) [inline code/modules/mob/living/simple_animal/friendly/farm_animals.dm]: And this is exactly the reason why it "constantly adds it's reagent to your blood".

**Comment by FantasticFwoosh** [comment]
I am eagerly awaiting squiggoths / goliath saddles / giant bloated savengers that havent been milked turned into huge aggressive war-beasts but yes in the meantime this is nice.

Will scavengers be on the list of xenobio gold core mobs? Given they have potential for bio-waste disposal on the station and medical application (feed dead monkeys for reliable heals)

**Comment by Iamgoofball** [comment]
MODS

**Comment by Zilenan91** [comment]
So are the ~~gretchins~~ scavengers ready to merge?

**Comment by RandomMarine** [comment]
This PR just reverted #17268

**Comment by Jordie0608** [comment]
Damnit cheri
madmelon

**Pull request merged**